com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -246,20 +247,49 @@ internal set
 
         private ulong m_LocalClientId;
 
+        internal Dictionary<ulong, NetworkClient> connectedClients = new Dictionary<ulong, NetworkClient>();
+
+        internal List<NetworkClient> connectedClientsList = new List<NetworkClient>();
+
+        internal List<ulong> connectedClientIds = new List<ulong>();
+
         /// <summary>
         /// Gets a dictionary of connected clients and their clientId keys. This is only populated on the server.
         /// </summary>
-        public readonly Dictionary<ulong, NetworkClient> ConnectedClients = new Dictionary<ulong, NetworkClient>();
+        public IReadOnlyDictionary<ulong, NetworkClient> ConnectedClients
+        {
+            get
+            {
+                Assert.IsTrue(IsServer, $"{nameof(ConnectedClients)} should only be accessed on server.");
```

Comment:
I don't think Assert should be used here - IIRC @JesseOlmer mentioned that Assert gets compiled out in production builds.

Reply:
Yeah my goal was to get them compiled out for perf. But now thinking about it I don't think we should care in this case. Will change to NotServerException.

Reply:
To elaborate, the problem with using exceptions inside properties is that it kills inlining for them.